elasticapm/contrib/opentracing/tracer.py

```diff
@@ -0,0 +1,96 @@
+
+import elasticapm
+from elasticapm import instrument, traces
+from elasticapm.conf import constants
+from elasticapm.contrib.opentracing.span import OTSpan, OTSpanContext
+from elasticapm.utils import compat, disttracing
+
+
+class Tracer(TracerBase):
+    _elasticapm_client_class = elasticapm.Client
+
+    def __init__(self, client_instance=None, config=None, scope_manager=None):
+        self._agent = client_instance or self._elasticapm_client_class(config=config)
+        if scope_manager and not isinstance(scope_manager, ThreadLocalScopeManager):
+            warnings.warn(
+                "Currently, the Elastic APM opentracing bridge only supports the ThreadLocalScopeManager. "
+                "Usage of other scope managers will lead to unpredictable results."
+            )
+        self._scope_manager = scope_manager or ThreadLocalScopeManager()
+        if self._agent.config.instrument:
+            instrument()
+
+    def start_active_span(
+        self,
+        operation_name,
+        child_of=None,
+        references=None,
+        tags=None,
+        start_time=None,
+        ignore_active_span=False,
+        finish_on_close=True,
+    ):
+        ot_span = self.start_span(
+            operation_name,
+            child_of=child_of,
+            references=references,
+            tags=tags,
+            start_time=start_time,
+            ignore_active_span=ignore_active_span,
+        )
+        scope = self._scope_manager.activate(ot_span, finish_on_close)
+        return scope
+
+    def start_span(
+        self, operation_name=None, child_of=None, references=None, tags=None, start_time=None, ignore_active_span=False
+    ):
+        if isinstance(child_of, OTSpanContext):
+            parent_context = child_of
+        elif isinstance(child_of, OTSpan):
+            parent_context = child_of.context
+        elif references and references[0].type == ReferenceType.CHILD_OF:
+            parent_context = references[0].referenced_context
+        else:
+            parent_context = None
+        transaction = traces.execution_context.get_transaction()
+        if not transaction:
+            trace_parent = parent_context.trace_parent if parent_context else None
+            transaction = self._agent.begin_transaction("opentracing", trace_parent=trace_parent)
+            transaction.name = operation_name
+            span_context = OTSpanContext(trace_parent=transaction.trace_parent)
+            ot_span = OTSpan(self, span_context, transaction)
+        else:
+            parent_span_id = (
+                parent_context.span.elastic_apm_ref.id
+                if parent_context and parent_context.span and not parent_context.span.is_transaction
+                else None
+            )
+            span = transaction.begin_span(operation_name, None, parent_span_id=parent_span_id)
+            trace_parent = parent_context.trace_parent if parent_context else transaction.trace_parent
```

Comment:
just to wrap my head around this, `parent_context.trace_parent` is not `None` when the span's parent is propagated from another service, right? if so, `parent_context.trace_parent` is not `None` means that `transaction` is `None` (because this is the first span of a service)?

another way of putting it: i actually don't understand this bit :see_no_evil: